src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -498,14 +510,38 @@ def convert_sweep_results(
                     qubit = qubits[ro_pulse.qubit]
                     discriminated_shots = self.classify_shots(i_vals, q_vals, qubit)
                     if execution_parameters.averaging_mode is AveragingMode.CYCLIC:
-                        discriminated_shots = np.mean(discriminated_shots, axis=0)
-                    result = execution_parameters.results_type(discriminated_shots)
+                        _, counts = np.unique(
+                            discriminated_shots, return_counts=True, axis=0
+                        )
+                        freqs = counts / np.shape(discriminated_shots)[0]
+                        result = execution_parameters.results_type(
+                            freqs, discriminated_shots
+                        )
+                    else:
+                        result = execution_parameters.results_type(discriminated_shots)
```

Comment:
Isn't this duplicated from above?

Reply:
I've added a method to avoid duplication :-)